bin/propolis-server/src/lib/spec/mod.rs

```diff
@@ -2,44 +2,196 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
-//! Helper functions for building instance specs from server parameters.
+//! Instance specs describe how to configure a VM and what components it has.
+//!
+//! This module defines an "internal" spec type for the server's instance
+//! initialization code to use. This spec type is not `Serialize` and is not
+//! meant to be sent over the wire in API requests or the migration protocol.
+//! This allows the internal representation to change freely between versions
+//! (so long as it can consistently be converted to and from the wire-format
+//! spec in the [`propolis_api_types`] crate); this, in turn, allows this
+//! representation to take forms that might otherwise be hard to change in a
+//! backward-compatible way.
 
```

Comment:
this is definitely something i'd been thinking i was missing in the last week or so. glad to see it!

the paragraph here points heavily towards `struct Spec`, but it's largely applicable other types as well, like `StorageDevice`. what do you think about changing the mention of
> an "internal" spec type ...

to something like

> "internal" spec types ... which are used elsewhere via an enclosing [\`Spec\`]"

?

seems worth emphasizing that the properties here are intended for all members of this module, but that `Spec` really is special in being where users will get references to anything else in here.

Reply:
Good call. I've updated the prose here (and fixed up a couple additional doc comments) in 87324c2.